dmutils/user.py

```diff
@@ -63,6 +64,32 @@ def has_role(self, role):
     def has_any_role(self, *roles):
         return any(self.has_role(role) for role in roles)
 
+    def is_part_of_team(self):
+        if self.has_role('buyer'):
+            if self.teams:
+                return True
+        return False
+
+    def is_team_lead(self):
+        if not self.is_part_of_team():
+            return True
```

Comment:
Should this be `False`? it seems to be saying the user is a team lead if they are not part of a team?

Or is this a way of giving the user team lead privileges when they have not yet joined a team? If so, using `is_team_lead()` will be prone to returning `True` when the user is a supplier/applicant, so it can't really be used on its own i.e. it will need to come after other role/permission checks, which is not clear by just reading the function name alone.

Reply:
returning False when not part of team